reactComponents/src/searchbar/redux/tokenSearchSlice.ts

```diff
@@ -50,58 +47,42 @@ const valueCleaner = (networkMap, exchangeMap) => {
   networkMap = Object.keys(omitBy(networkMap, (b) => !b));
   exchangeMap = Object.keys(omitBy(exchangeMap, (b) => !b));
 
-
   // Returns the processed values of "networkMap" and "exchangeMap".
   return [networkMap, exchangeMap];
 };
 
+export const searchTokenPairs = createAsyncThunk('token/search', async (dataProp: any, thunkAPI: any) => {
+  try {
+    const { networkMap, exchangeMap } = thunkAPI.getState();
+    const { searchString, networks } = dataProp;
+    let processedNetworks;
+    let processedExchanges;
+    const pairSearchTimestamp = new Date().getTime();
+
+    // Dispatches "setPairSearchTimestamp".
+    thunkAPI.dispatch(setPairSearchTimestamp(pairSearchTimestamp));
+
+    // Runs the function handling the cleaning of the properties from their values indicating if they are enabled or not.
+    [processedNetworks, processedExchanges] = valueCleaner(networkMap, exchangeMap);
+
+    // Runs the function handling the management of the "All" value selected by the user.
+    [processedNetworks, processedExchanges] = allValueHandler(
```

Comment:
i'm sure this logic works, but would a more functional approach be safer/easier to follow?

Here you are using the return values of a function call that were arguments to the function to change the value of the same variable. It makes me uncomfortable